tsfile/src/main/java/org/apache/iotdb/tsfile/write/record/Tablet.java

```diff
@@ -121,13 +121,11 @@ public void addValue(String measurementId, int rowIndex, Object value) {
     IMeasurementSchema measurementSchema = schemas.get(indexOfValue);
 
     if (measurementSchema.getType().equals(TSDataType.VECTOR)) {
-      for (int i = 0; i < measurementSchema.getValueMeasurementIdList().size(); i++) {
-        TSDataType dataType = measurementSchema.getValueTSDataTypeList().get(i);
-        addValueOfDataType(dataType, rowIndex, measurementIndex.get(measurementId), value);
-      }
+      int indexInVector = measurementSchema.getMeasurementIdColumnIndex(measurementId);
+      TSDataType dataType = measurementSchema.getValueTSDataTypeList().get(indexInVector);
+      addValueOfDataType(dataType, rowIndex, indexOfValue + indexInVector, value);
```

Comment:
We need to consider when `value == null`, init the `BitMaps` in tablet if needed, set an irrelevant value in the array, and mark this position of bitMap.

Reply:
Fixed by myself...